lib/auth/onlinesaml.js

```diff
@@ -16,6 +17,7 @@ module.exports = function(client, httpOpts, cb) {
   var waterfall = [];
   
   if (!client.FedAuth || !client.rtFa){
+    assert.ok(client.federatedAuthUrl, "Federated Auth URL is required when using onlinesaml");
```

Comment:
A little unsure of using assert outside of a test, since it'll throw, rather than returning a nice error message. Not to be fixed for now, but consider in a subsequent P/R maybe!?